packages/services/api/src/modules/audit-logs/providers/audit-logs-manager.ts

```diff
@@ -0,0 +1,339 @@
+        },
+      });
+    }
+
+    const getPresignedUrl = await client.fetch([endpoint, bucket, key].join('/'), {
+      method: 'GET',
+      aws: {
+        signQuery: true,
+      },
+    });
+
+    if (!getPresignedUrl.ok) {
+      return Promise.resolve({
+        error: {
+          message: 'Failed to get the pre-signed URL',
+        },
+      });
+    }
+
+    return Promise.resolve({
+      ok: {
+        url: getPresignedUrl.url,
+      },
+    });
+  }
+
+  async exportAndSendEmail(
+    organizationSlug: string,
+    filter: { startDate?: Date | null; endDate?: Date | null },
+  ): Promise<{ ok: { url: string } } | { error: { message: string } }> {
+    const formattedStartDate = filter?.startDate ? new Date(filter.startDate) : undefined;
+    const formattedEndDate = filter?.endDate ? new Date(filter.endDate) : undefined;
+
+    const currentUser = await this.session.getViewer();
+    const currentUserEmail = currentUser.email;
+
+    const getAllAuditLogs = await this.getPaginatedAuditLogs(
+      organizationSlug,
+      { first: 1000, after: 0 },
+      { startDate: formattedStartDate, endDate: formattedEndDate },
+    );
+
+    if (!getAllAuditLogs || !getAllAuditLogs.data || getAllAuditLogs.data.length === 0) {
+      return Promise.resolve({
+        error: {
+          message: 'No audit logs found',
+        },
+      });
+    }
+
+    const data = getAllAuditLogs.data.map(log => {
+      const { id, event_time, user_id, user_email, organization_id, event_action, metadata } = log;
+      return {
+        ID: id,
+        Type: event_action,
+        CreatedAt: event_time,
+        UserID: user_id,
+        UserEmail: user_email,
+        OrganizationID: organization_id,
+        Metadata: JSON.stringify(metadata),
+      };
+    });
+
+    const columns = [
+      'ID',
+      'Type',
+      'CreatedAt',
+      'UserID',
+      'UserEmail',
+      'OrganizationID',
+      'Metadata',
+    ];
+
+    const currentDate = new Date().toLocaleDateString();
+    const currentTime = new Date().toLocaleTimeString();
+
+    const customHeader = [
+      [`Hive Audit Logs for Organization: ${organizationSlug}`],
+      [`Date: ${currentDate}`],
+      [`Time: ${currentTime}`],
+      [`User: ${currentUserEmail}`],
+      [''], // Blank row
+      [''], // Another blank row to separate sections
+    ];
+
+    // Convert data to CSV format
+    const csvData = await new Promise<string>((resolve, reject) => {
+      stringify(
+        data,
+        {
+          header: true,
+          columns,
+        },
+        (err, output) => {
+          if (err) {
+            reject(err);
+          } else {
+            const csv = customHeader.join('\n') + '\n' + output;
+            resolve(csv);
+          }
+        },
+      );
+    });
+
+    const s3Storage = this.s3Config[0];
+    const { endpoint, bucket, client } = s3Storage;
+    const key = `audit-logs/${organizationSlug}/${new Date().toISOString()}.csv`;
```

Comment:
TODO @TuvalSimha: use a date string not containing special characters like `:` and spaces. E.g. YYYY_MM_DD_HOUR_MINUTE_SECOND